sunpy/coordinates/utils.py

```diff
@@ -265,3 +265,69 @@ def coordinates(self, points=None):
                         obstime=self.obstime,
                         observer=self.observer,
                         frame=Heliocentric).transform_to(self.start_frame)
+
+
+def get_rectangle_coordinates(bottom_left, *, top_right = None, width: u.deg = None, height: u.deg = None):
+    """
+    Specify a rectangular region of interest in spherical world coordinates.
+
+    Parameters
+    ----------
+    bottom_left : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The bottom_left coordinate of the rectangle. If a `SkyCoord` it can
+        have shape ``(2,)`` and also define ``top_right``.
+
+    top_right : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The top_right coordinate of the rectangle. Can only be omitted if
+        ``bottom_left`` has shape ``(2,)``.
+
+    width : `~astropy.units.Quantity`
+        The width of the rectangle.
+        Must be omitted if ``bottom_left`` has shape ``(2,)`` or ``top_right`` is provided.
+
+    height : `~astropy.units.Quantity`
+        The height of the rectangle.
+        Must be omitted if ``bottom_left`` has shape ``(2,)`` or ``top_right`` is provided.
+
+    Returns
+    -------
+    bottom_left : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The bottom_left coordinate of the rectangular region of interest in spherical world coordinates.
+
+    top_right : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`    
+        The top_right coordinate of the rectangular region of interest in spherical world coordinates.
+    """
+    if bottom_left.shape == (2,) and any((width, height, top_right)):
+        raise ValueError("Invalid input, if bottom_left.shape == (2,) \
+                          other parameters should not be passed.")
+
+    if all((width, height, top_right)):
+        raise ValueError("Invalid input, width, height and top_right \
+                          parameters should not be passed simultaneously.")
+
+    if top_right is None and bottom_left.shape != (2,) and not all((width, height)):
+        raise ValueError("Invalid input, either bottom_left and top_right \
+                         or bottom_left and height and width should be provided.")
+
+    if (not isinstance(bottom_left, (SkyCoord, BaseCoordinateFrame)) and 
+        not isinstance(top_right, (None, SkyCoord, BaseCoordinateFrame))):
```

Comment:
Not a big deal, but a duck-typing approach would be nicer than explicitly checking the type.  That is, checking for the existence of a `.transform_to()` method and a `.spherical` property, which is all that you use.